src/west/manifest.py

```diff
@@ -183,6 +189,18 @@ def _load(self, data, sections):
 
         manifest = data.get('manifest')
 
+        url = config.get('self', 'remote', fallback='origin')
+        name = posixpath.basename(urlparse(url).path)
+        revision = config.get(name, 'revision', fallback='master')
+        path = os.path.join(name)
+
+        self_tag = manifest.get('self')
+        if self_tag:
+            path = self_tag['path']
+
+        project = SpecialProject(name, revision=revision, path=path, url=url)
+        projects.append(project)
```

Comment:
The convention that the manifest project is `projects[0]` is a good one. It should be added to the pydoc so users can rely on it as part of the API.